core-services/workflow/src/main/java/org/silverpeas/core/workflow/engine/QuestionRequest.java

```diff
@@ -0,0 +1,136 @@
+      newStep.setUserId(event.getUser().getUserId());
+      newStep.setAction("#question#");
+      newStep.setActionDate(event.getActionDate());
+      newStep.setUserRoleName(event.getUserRoleName());
+      if (event.getResolvedState() != null) {
+        newStep.setResolvedState(event.getResolvedState().getName());
+      }
+      // To be processed
+      newStep.setActionStatus(0);
+      newStep.setProcessInstance(processInstance);
+
+      // add the new newStep to the processInstance
+      getHistoryStepRepository().save((HistoryStepImpl) newStep);
+
+      step.set(newStep);
+      return newStep;
+    });
+
+    Transaction.performInOne(()-> {
+      UpdatableProcessInstance processInstance = getProcessInstanceRepository().getById(id);
+
+      // Do workflow stuff
+      try {
+        processEvent(processInstance, step.get().getId());
+        getProcessInstanceRepository().save((ProcessInstanceImpl) processInstance);
+      } catch (WorkflowException we) {
+        saveError(processInstance, event, step.get(), we);
+
+        throw new WorkflowException("WorkflowEngineThread.process",
+            "workflowEngine.EX_ERR_PROCESS_ADD_TASKSAVED_REQUEST", we);
+      }
+
+      return null;
+    });
+  }
+
+  /**
+   * Method declaration
+   */
+  private void processEvent(UpdatableProcessInstance instance, String stepId)
+      throws WorkflowException {
+
+    // Create a history step for the question
+    UpdatableHistoryStep step = (UpdatableHistoryStep) instance.getHistoryStep(stepId);
+    // only to set the current step to that step
+    instance.updateHistoryStep(step);
+
+    // add the question
+    String question;
+    try {
+      question = (String) event.getDataRecord().getField("Content").getObjectValue();
+    } catch (FormException fe) {
+      throw new WorkflowException("WorkflowEngineThread.process", "workflowEngine.EXP_UNKNOWN_ITEM",
+          fe);
+    }
+    State state = instance.addQuestion(question, event.getStepId(), event.getResolvedState(), event.
+        getUser());
+
+    // add the state that is discussed in the list of active states
+    instance.addActiveState(state);
+
+    // change the action status of the step
+    step.setActionStatus(1); // Processed
+    instance.updateHistoryStep(step);
+
+    // get the last participant for the discussed state
+    Participant participant = instance.getParticipant(state.getName());
+
+    // Get the task manager
+    TaskManager taskManager = WorkflowHub.getTaskManager();
+
+    // Assign task to this participant
+    Task task = taskManager.createTask(participant, instance);
+    taskManager.assignTask(task, participant.getUser());
+
+    // Declare this user as a working user in instance
+    instance.addWorkingUser(participant.getUser(), state, participant.getUserRoleName());
+
+    // change the action status of the step
+    // Affectations done
+    step.setActionStatus(2);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 2 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)